Description: ignore certificate expiries when verifying signatures
 The UEFI implementation explicitly ignores all errors due to expired (or
 not yet valid) signatures.  Ensure that sbverify behaves compatibly.
Author: Steve Langasek <steve.langasek@canonical.com>
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1234649.
Last-Update: 2013-10-03

Index: sbsigntool-0.6/src/sbverify.c
===================================================================
--- sbsigntool-0.6.orig/src/sbverify.c
+++ sbsigntool-0.6/src/sbverify.c
@@ -206,6 +206,13 @@
 		if (cert_in_store(ctx->current_cert, ctx))
 			status = 1;
 	}
+	/* UEFI doesn't care about expired signatures, so we shouldn't either. */
+	else if (err == X509_V_ERR_CERT_HAS_EXPIRED ||
+			err == X509_V_ERR_ERROR_IN_CERT_NOT_BEFORE_FIELD ||
+			err == X509_V_ERR_CERT_NOT_YET_VALID ||
+			err == X509_V_ERR_ERROR_IN_CERT_NOT_AFTER_FIELD) {
+		status = 1;
+	}
 
 	return status;
 }
